tests/models/test_default_evaluator.py

```diff
@@ -499,3 +502,160 @@ def test_gen_multiclass_roc_curve():
 
     expected_auc = [0.75, 0.75, 0.3333]
     assert np.allclose(results.auc, expected_auc, rtol=1e-3)
+
+
+def test_evaluate_custom_metric_incorrect_return_formats():
```

Comment:
The test looks thorough, but it could be improved by using the parametrize method from pytest: https://docs.pytest.org/en/6.2.x/example/parametrize.html#parametrizing-conditional-raising

Reply:
Oh that's interesting, I didn't know about the parametrize method. But I wonder if I should use it here because this test only contain cases that raise exceptions. And seems like from the pytest documentation referenced above parametrize method is used in tests that contain cases that raise and don't raise exceptions. Perhaps did you mean I should rewrite my tests here so that they contain both exception-raising and exception-not-raising cases?

Reply:
No need to rewrite if it's difficult to factor them out. Leave them as is is fine.